### ISSUE TITLE
Compile error on shfl_down_sync

### ISSUE THREAD

**Issue by koenvos** [JuliaGPU/CUDA.jl#834]
The code below stopped working in CUDA 3.
```
a = CUDA.ones(32)
function kernel(a)
    w = 16
    while w > 0
        a += shfl_down_sync(0xffffffff, a, w)
        w >>= 1
    end
    return nothing
end
@cuda threads=1 kernel(a)
```
Gives
```
ERROR: GPU compilation of kernel kernel(CuDeviceVector{Float32, 1}) failed
KernelError: kernel returns a value of type `Union{}`
```

Setup:
```
CUDA v3.0.1
GPUCompiler v0.11.3
GPUArrays v0.11.3
LLVM v3.6.0
```
```
versioninfo():
Julia Version 1.6.0
Commit f9720dc2eb (2021-03-24 12:55 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = "C:\Program Files\Microsoft VS Code\Code.exe"
  JULIA_NUM_THREADS = 6
```

**Comment by maleadt** [comment]
Are you sure? You're trying to shuffle the actual array here. Didn't you intend to to `b = a[1]` and shuffle `b`?

**Comment by koenvos** [comment]
Oops, you're right of course.
Let me try again to reduce my problem to a minimum example :)
